### PR TITLE
Improve input fields a11y

### PR DESCRIPTION
**Pull request by rafawendel**
## What type of PR is this? (check all applicable)

- [X] Feature

## Description
Redash makes wide use of `<input>` fields as search fields and alike functionalities. Most of these are accessible by context, but lack systematic or explicit accessibility, which is the intent of this PR. Namely:

1. Add fallback `aria-label`s where applicable
2. Explicitly link `<label>`s and fields whenever possible
3. Replace meaningless adjunct text with `<label>` with no visual regressions
4. Introduce `aria-labelledby` and/or `aria-describedby` to edge cases

### Note
Ant Design `<Search>` and `<Dropdown>` elements make use of `<input>` in a controversial fashion, which will be addressed in a further PR.

## Related Tickets & Documents
See a11y epic.

### CODE REVIEW

File: client/app/components/items-list/components/Sidebar.jsx [hunk 1]
```diff
@@ -29,7 +29,13 @@ export function SearchInput({ placeholder, value, showIcon, onChange }) {
   const InputControl = showIcon ? Input.Search : Input;
   return (
     <div className="m-b-10">
-      <InputControl className="form-control" placeholder={placeholder} value={currentValue} onChange={onInputChange} />
+      <InputControl
+        className="form-control"
+        placeholder={placeholder}
+        value={currentValue}
+        aria-label="Search" // TODO: consider receiving as prop (depends on parent render)
```
Comment (kravets-levko) [inline client/app/components/items-list/components/Sidebar.jsx:36]: I think "Search" is good enough here

File: client/app/components/queries/SchemaBrowser.jsx [hunk 2]
```diff
@@ -232,6 +232,7 @@ export default function SchemaBrowser({
         <Input
           className="m-r-5"
           placeholder="Search schema..."
+          aria-label="Schema"
```
Comment (kravets-levko) [inline client/app/components/queries/SchemaBrowser.jsx:235]:
```suggestion
          aria-label="Search schema"
```

File: client/app/components/queries/editor-components/databricks/DatabricksSchemaBrowser.jsx [hunk 3]
```diff
@@ -84,6 +84,7 @@ export default function DatabricksSchemaBrowser({
         <Input
           className={isDatabaseSelectOpen ? "database-select-open" : ""}
           placeholder="Filter tables & columns..."
+          aria-label="Schema"
```
Comment (kravets-levko) [inline client/app/components/queries/editor-components/databricks/DatabricksSchemaBrowser.jsx:87]:
```suggestion
          aria-label="Search schema"
```
...or whatever fits better (on your opinion). IMHO "Schema" is too generic and doesn't reflect the purpose of the input

**Review by kravets-levko — APPROVED** [review]
In general looks good, just few comments (not blockers)

File: client/app/components/queries/editor-components/databricks/DatabricksSchemaBrowser.jsx (hunk 3 above)
Comment (rafawendel) [reply client/app/components/queries/editor-components/databricks/DatabricksSchemaBrowser.jsx:87]: SGTM!

**Pull request merged**